ckanext/versioning/logic/action.py

```diff
@@ -1,22 +1,68 @@
 # encoding: utf-8
+import datapackage
 import difflib
 import json
 import logging
 import re
 from datetime import datetime
 
+
 from ckan import model as core_model
+from ckan_datapackage_tools import converter
 from ckan.logic.action.get import package_show as core_package_show
 from ckan.logic.action.get import resource_show as core_resource_show
+from ckan.logic.action.create import package_create as core_package_create
+from ckan.logic.action.update import package_update as core_package_update
 from ckan.plugins import toolkit
 from sqlalchemy.exc import IntegrityError
-
 from ckanext.versioning.logic import helpers as h
 from ckanext.versioning.model import DatasetVersion
+from metastore.backend.gh import GitHubStorage
 
 log = logging.getLogger(__name__)
 
 
+def _get_github_backend():
```

Comment:
I would make this `_get_metastore_backend` and not assume we are using GitHub. The whole point of `metastore-lib` is to abstract that away. Just use `metastore.create_metastore` and pass it two config options: `backend_type` and `backend_config`. The latter can be a dict - I know CKAN allows that. This allows us to use the GitHub storage in production and the Filesystem storage locally / in testing, and also any other backend in the future without changing the implementation.

Reply:
Hmm... good question. We can surely use a disk-based storage as you did, and then just delete that directory on test `teardown`. I didn't try but it should work. 

It does bring up a question of whether or not we want the backend object somehow persisted in memory e.g. on the Plugin object or even in the module level. Maybe if CKAN has something like request-locals we can add another function to `ckanext/versioning/common.py` that calls `get_metastore_backend` and caches the backend object as a request-local variable. That will allow us to have a single backend object as long as we are on the same request context, which makes sense. I'm not sure it's worth it though, and will not play nice with tests that do not run inside a request context.